test/e2e/basic_test.go

```diff
@@ -334,3 +351,14 @@ func findPodInPodStats(summary *v1alpha1.Summary, pod *v1.Pod) (int, error) {
 	}
 	return -1, fmt.Errorf("failed to find pod \"%s/%s\" in the slice of pod stats", pod.Namespace, pod.Name)
 }
+
+// findPodInPodStats returns the index of the specified pod in the .pods field of the specified PodList object.
+// It returns error if the pod doesn't exist in the podlist
+func findPodInPods(pods *v1.PodList, pod *v1.Pod) error {
```

Comment:
Nit: You can return a bool here instead of an error and simplify the asserts based on this.
It also seem like when we get the pod list all we ever want to do is check if the pod we have is running.
Maybe this can just be part of the framework? e.g. `assert.Assert(t, f.PodIsRunning(pod))`

Reply:
I tried to keep this to the semantics of findPodInPodStats. If you're fine with the change, I'm all for it.

Reply:
Yeah since we're changing I think it's good.